lib/stdlib/test/shell_SUITE.erl

```diff
@@ -428,6 +428,31 @@ records(Config) when is_list(Config) ->
     [{error,invalid_filename}] = scan(<<"rr({foo}).">>),
     [[]] = scan(<<"rr(\"not_a_file\").">>),
 
+    %% load record from archive
+    _ = code:purge(test),
+    true = code:delete(test),
+
+    PrivDir = proplists:get_value(priv_dir, Config),
+    AppDir = filename:join(PrivDir, "test_app"),
+    ok = file:make_dir(AppDir),
+    AppEbinDir = filename:join(AppDir, "ebin"),
+    ok = file:make_dir(AppEbinDir),
+
+    ok = file:write_file(Test, Contents),
+    {ok, test} = compile:file(Test, [{outdir, AppEbinDir}]),
+
+    Ext = init:archive_extension(),
+    Archive = filename:join(PrivDir, "test_app" ++ Ext),
+    {ok, _} = zip:create(Archive, ["test_app"], [{compress, []}, {cwd, PrivDir}]),
+
+    ArchiveEbinDir = filename:join(Archive, "test_app/ebin"),
+    true = code:add_path(ArchiveEbinDir),
```

Comment:
The code path is a global state that needs to be reset when the testcase finishes. Otherwise it will interact and possibly fail other testcases. I noticed this as the shell_docs_SUITE testcases started to fail. I suggest that you save the path in init_per_suite and reset it again in end_per_suite.

Reply:
ah, thanks, will do